doc/reference.rst

```diff
@@ -338,6 +338,148 @@ use-dependency-links, default: true
 versions, default 'versions'
   The name of a section that contains :ref:`version pins <pinned-versions>`.
 
+Configuration file syntax
+=========================
+
+Buildout configurations use an `INI file format
+<https://en.wikipedia.org/wiki/INI_file>`_.
+
+A configuration is a collection of named sections containing named
+options.
+
+Section names
+-------------
+
+A section begins with a section and and, optionally, a condition in
+square braces (``[`` and ``]``).
+
+A name can consist of any characters other than whitespace, square
+braces, curly braced (``{`` or ``}``), point signs (``#``), colons
+(``:``) or semi-colons (``;``).  The name may be surrounded by leading
+and trailing whitespace, which is ignored.
+
+An optional condition is separated from the name by a colon and is a
+Python expression. It may not contain a pound sign or semi-colon.  See
```

Comment:
Actually, I'm trying to break myself of that habit. :)  But I'll add it here because this is text and because you mentioned it.